Modules/_ctypes/stgdict.c

```diff
@@ -681,24 +719,116 @@ PyCStructUnionType_update_stgdict(PyObject *type, PyObject *fields, int isStruct
             if (pair == NULL) {
                 return -1;
             }
+            if (!PyArg_ParseTuple(pair, "UO|i", &name, &desc, &bitsize)) {
+                PyErr_SetString(PyExc_TypeError,
+                    "'_fields_' must be a sequence of (name, C type) pairs");
+                Py_XDECREF(pair);
```

Comment:
Why not use `Py_DECREF()`?  Don't we already know that `pair` isn't NULL?

Reply:
Good point, I'll change that.